web/text_layer_builder.js

```diff
@@ -83,37 +91,83 @@ class TextLayerBuilder {
    * @param {number} [timeout] - Wait for a specified amount of milliseconds
    *                             before rendering.
    */
-  render(timeout = 0) {
-    if (!(this.textContent || this.textContentStream) || this.renderingDone) {
+  render(viewport, timeout = 0) {
+    if (!(this.textContent || this.textContentStream)) {
       return;
     }
-    this.cancel();
 
-    this.textDivs.length = 0;
-    this.highlighter?.setTextMapping(this.textDivs, this.textContentItemsStr);
-    this.accessibilityManager?.setTextMapping(this.textDivs);
-
-    const textLayerFrag = document.createDocumentFragment();
-    this.textLayerRenderTask = renderTextLayer({
-      textContent: this.textContent,
-      textContentStream: this.textContentStream,
-      container: textLayerFrag,
-      viewport: this.viewport,
-      textDivs: this.textDivs,
-      textContentItemsStr: this.textContentItemsStr,
-      timeout,
-    });
-    this.textLayerRenderTask.promise.then(
-      () => {
-        this.textLayerDiv.append(textLayerFrag);
-        this.#finishRendering();
-        this.highlighter?.enable();
-        this.accessibilityManager?.enable();
-      },
-      function (reason) {
-        // Cancelled or failed to render text layer; skipping errors.
+    const scale = viewport.scale * (globalThis.devicePixelRatio || 1);
```

Comment:
In order to reduce the number of parameters that need to be passed to the `src/display/text_layer.js` code, could we perhaps track the `scale` *internally* in that code instead rather than handling that manually on the `web/`-side of things?